Avoid casting to `NSError` in favor of `WordPressApiError`

mokagio opened the pull request:
See discussion at
https://github.com/wordpress-mobile/WordPressKit-iOS/pull/777#discussion_r1552498217 with @crazytonyli

### Testing Details

See green CI.

---

- [ ] Please check here if your pull request includes additional test coverage. — N.A.
- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary. — N.A.

mokagio committed:
Avoid casting to `NSError` in favor of `WordPressApiError`

See discussion at
https://github.com/wordpress-mobile/WordPressKit-iOS/pull/777#discussion_r1552498217
with @crazytonyli

crazytonyli commented on Sources/WordPressKit/Services/JetpackServiceRemote.swift:
```diff
@@ -84,11 +84,17 @@ public class JetpackServiceRemote: ServiceRemoteWordPressComREST {
+            let jetpackError = JetpackInstallError(
+                title: endpointError.localizedDescription,
+                code: endpointError.errorCode,
+                key: endpointError.errorUserInfo[WordPressComRestApi.ErrorKeyErrorCode] as? String
```
I believe you can do `endpointError.apiErrorCode` instead of using `errorUserInfo`.

crazytonyli commented on Sources/WordPressKit/Services/ActivityServiceRemote.swift:
```diff
@@ -152,18 +152,15 @@ open class ActivityServiceRemote: ServiceRemoteWordPressComREST {
 
-                                    let status = RewindStatus(state: .unavailable)
-                                    success(status)
+                                    failure(endpointError)
```
The original code returns `error`. But there it's changed to `endpointError`. I don't think we should change it here?

mokagio committed:
Rename `castToEndpointErrorWitCode` to `castedToEndpointErrorWitCode`

mokagio committed:
Restore returning `error` instead of cast `endpointError`

Thanks for the hint @crazytonyli.
See
https://github.com/wordpress-mobile/WordPressKit-iOS/pull/779/files#r1552634468

mokagio committed:
Use better API to set error key code in `JetpackServiceRemote`

crazytonyli commented on Sources/WordPressKit/Services/ActivityServiceRemote.swift:
```diff
@@ -152,18 +152,15 @@ open class ActivityServiceRemote: ServiceRemoteWordPressComREST {
+                                    guard error.castedToEndpointErrorWitCode(.preconditionFailure) != nil else {
+                                        success(RewindStatus(state: .unavailable))
                                         return
                                     }
```
My brain can't process these two guard statements. But, are their logic changed?

crazytonyli replied:
I noticed an unit test failed, I think this guard change here might be the cause?
